### ISSUE TITLE
Suggestion: use UAX #14 Unicode Line Breaking Algorithm for determining legal breaks before running K&P

### ISSUE THREAD

**Issue by shermp** [baskerville/paragraph-breaker#1]
Hi baskerville

I don't know if you've thought about this yet, as I'm aware that this project is in it's very early stages, but as the title says, I suggest maybe implementing the UAX #14 algorithm to find all the legal breaks in a paragraph.

Once all the legal breaks have been found, one can then use K&P to determine the best breakpoints to use.

The reason I suggest this is that Unicode have done all the hard work of figuring out where breaks are allowed, depending on language used. It would save the situation that I've seen the Koreader devs encounter with crengine, with having to add new rules as readers of languages other than english encounter issues.

Of course, you may have already thought of this, and what I've just said is completely redundant..

I'm looking forward to an (X)HTML renderer that can do K&P line breaking. Thank you for taking on the challenge!

**Comment by baskerville** [comment]
Thanks, there's already an implementation of [UAX #14](https://github.com/google/xi-editor/tree/master/rust/unicode).

**Comment by shermp** [comment]
Yeah, I figured there would be. I imagine an implementation of UAX #14 has been done in many languages.

I guess I wasn't sure whether you were planning on rolling your own rules or not, as I have seen some (experimental) implementations of K&P do just that.
